backend/src/main/java/com/example/backend/filter/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,74 @@
+package com.example.backend.filter;
+
+import java.io.IOException;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+import com.example.backend.service.JwtService;
+
+import io.jsonwebtoken.Claims;
+
+@Component
+@RequiredArgsConstructor
+// validates the access token before any request that needs authorization
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+    private final JwtService jwtService;
+    private final UserDetailsService userDetailsService;
+
+    @Override
+    protected void doFilterInternal(
+            @NonNull HttpServletRequest request,
+            @NonNull HttpServletResponse response,
+            @NonNull FilterChain filterChain) throws ServletException, IOException {
+        final String authHeader = request.getHeader("Authorization");
+
+        // authentication header should be of the form "Bearer {accessToken}"
+        if (authHeader == null || !authHeader.startsWith("Bearer ")) {
```

Comment:
Isn't there a better way to do this than hardcoding the value of "Bearer "?
You can use a constant where you will store that value.

Reply:
asa e protocolu ma ce tot vr =))